sql/src/test/java/io/crate/integrationtests/LicenseViolationITest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.integrationtests;
+
+import io.crate.action.sql.SQLActionException;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 3)
```

Comment:
I think this could be flaky due to master only or client only nodes. I'm also wondering if this kind of integration test is even necessary. 
It looks like it is testing whether the `LicenseService` is handling cluster state events (node changes directly) - but the assertions themselves are based on whether the logic in the planner detect correctly whether `create table` is allowed or not on an expired license. 

Not sure if this additional indirection is beneficial.

Maybe we could replace this with a `LicenseService` unit test that checks the cluster state update handling is working.
Or we could instead extend `testLicenseIsAvailableInClusterStateAfterSetLicense` so that it shows that the `maxNumberOfNodes` is updated accordingly.